src/box/memtx_engine.c

```diff
@@ -78,6 +78,117 @@ enum {
 	MAX_TUPLE_SIZE = 1 * 1024 * 1024,
 };
 
+enum {
+	/* Space _schema recover complete index */
+	SCHEMA_RECOVER_COMPLETE_ID,
+	/* Space _collation recover complete index */
+	COLLATION_RECOVER_COMPLETE_ID,
+	/* Space _space recover complete index */
+	SPACE_RECOVER_COMPLETE_ID,
+	/* Space _index recover complete index */
+	INDEX_RECOVER_COMPLETE_ID,
+	/* Space _func recover complete index */
+	FUNC_RECOVER_COMPLETE_ID,
+	/* Space _user recover complete index */
+	USER_RECOVER_COMPLETE_ID,
+	/* Space _priv recover complete index */
+	PRIV_RECOVER_COMPLETE_ID,
+	/* Space _cluster recover complete index */
+	CLUSTER_RECOVER_COMPLETE_ID,
+	/* End of enum */
+	RECOVER_COMPETE_MAX_ID,
```

Comment:
I see only one method of check that a space with some ID was fully loaded - if a read a row with space_id > ID. Thus we can ensure that all system spaces was fully loaded if we read a row with space_id = BOX_SYSTEM_ID_MAX + 1. And if we read such a row - that means that we have read ALL system spaces and thus we should not care about each of them individually.